source/s3_default_meta_request.c

```diff
@@ -18,6 +18,9 @@ enum aws_s3_meta_request_default_state {
 struct aws_s3_meta_request_default {
     struct aws_s3_meta_request base;
 
+    bool is_get_request;
```

Comment:
Isn't there already a request type?

Reply:
Also, looking at how this is used, and request_type for that manner, why don't we just set the vtable up differently based on the request type when we initialize it, and then never branch?

Reply:
We could do that!

Reply:
Actually, we only do one branch on this right now to add an additional flag for the request.  Was wondering if this might be overkill? :-/  (Although I do like the idea.)